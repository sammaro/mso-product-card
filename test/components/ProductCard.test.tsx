import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product1 } from '../data/products';

const { act } = renderer;

describe('ProductCard', () => {
  test('should display the product card', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <h1>ProductCard</h1>}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should increment counter', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {({ count, increaseBy }) => (
          <>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}>Increase</button>
          </>
        )}
      </ProductCard>
    );

    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[1].props.onClick();
    });
    tree = wrapper.toJSON();
    expect((tree as any).children[0].children[0]).toBe('1');
  });
});
