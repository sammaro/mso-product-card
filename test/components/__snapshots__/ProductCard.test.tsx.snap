// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should display the product card 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
</div>
`;

exports[`ProductCard should increment counter 1`] = `
<div
  className="productCard undefined"
>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    Increase
  </button>
</div>
`;
